public/js/actions/breakpoints.js

```diff
@@ -22,15 +23,51 @@ function enableBreakpoint(location: Location) {
   return addBreakpoint(location);
 }
 
+function isOriginalLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  return isOriginal(source.toJS());
+}
+
+function getGeneratedLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  if (isOriginal(source.toJS())) {
+    return getGeneratedSourceLocation(source.toJS(), location);
+  }
+
+  return location;
+}
+
 function _breakpointExists(state, location: Location) {
+  location = getGeneratedLocation(state, location);
   const currentBp = getBreakpoint(state, location);
   return currentBp && !currentBp.get("disabled");
 }
 
 function _getOrCreateBreakpoint(state, location, condition) {
+  location = getGeneratedLocation(state, location);
   return getBreakpoint(state, location) || fromJS({ location, condition });
 }
 
+function getBreakpointText({
+  breakpoint, getState, getTextForLine, actualLocation, location }) {
```

Comment:
I see. I think the tension is with `actualLocation`. If we're sourcemapping, we know the server doesn't slide breakpoints so we can just use the original location. However, Chrome's server does slide even with sourcemaps. I think you should always use `actualLocation`, sourcemap that *back* to an original location and pass it here as a single location.

This stuff is so confusing.... :p